src/wallet/db.cpp

```diff
@@ -672,6 +672,12 @@ bool CWalletDBWrapper::Backup(const std::string& strDest)
                     pathDest /= strFile;
 
                 try {
+                    if (fs::equivalent(pathSrc, pathDest)) {
```

Comment:
Good that you've included this in the try-except block, since `boost::filesystem::equivalent()` can throw, but the generic catch behaviour below would hide the precise error. You could return a more precise error by:

- using the `equivalent(const path& p1, const path& p2, system::error_code& ec)` form (http://www.boost.org/doc/libs/1_46_0/libs/filesystem/v3/doc/reference.html#equivalent) and checking the returned `error_code`
- testing the error code in the catch block below
- having two try-catch blocks

Perhaps a little bit overkill since the only way that `boost::filesystem::equivalent()` can throw is if the source `wallet.dat` file doesn't exist or isn't a regular file/directory/symlink. But something to consider.

Reply:
>  but the generic catch behaviour below would hide the precise error.

It does dumps the error message

> Perhaps a little bit overkill since the only way that boost::filesystem::equivalent() can throw is if the source wallet.dat file doesn't exist or isn't a regular file/directory/symlink. 

This was also my thought. If equivalant() fails we have bigger problems and a generic error is sufficient.